lib/omniauth/strategies/okta.rb

```diff
@@ -51,8 +51,9 @@ class Okta < OmniAuth::Strategies::OAuth2
 
       def access_token
         ::OAuth2::AccessToken.new(client, oauth2_access_token.token, {
-          :expires_in => oauth2_access_token.expires_in,
-          :expires_at => oauth2_access_token.expires_at
+          :refresh_token => oauth2_access_token.refresh_token,
+          :expires_in    => oauth2_access_token.expires_in,
+          :expires_at    => oauth2_access_token.expires_at
```

Comment:
sorry to get to this old commit, but I need to know how this functions is working, in line 50 it's aliased to `oauth2_access_token` so that means this function is calling itself endlessly. so how is this working?

Reply:
What is happening is the base class is defining an accessor `access_token` [here](https://github.com/omniauth/omniauth-oauth2/blob/master/lib/omniauth/strategies/oauth2.rb#L44) so when we get to the `alias` on line 50 it aliases the method from the base class to `oauth2_access_token` (since we haven't overwritten it yet) then we define a new `access_token` method and when we call `oauth2_access_token` it is actually referencing the method defined on the base class, not the method defined in this class. Ruby is bananas 🤯

Reply:
ooooh, I see, makes sense now. thanks for taking the time to explain it. have a nice weekend :wave: